### ISSUE TITLE
myelin::ir::tensor_t*& myelin::ir::operand_t::tensor()

### ISSUE THREAD

**Issue by quintetoy** [NVIDIA/TensorRT#1541]
I have got a problem like this,I was running pointpillars model-- pillarscatter, and when i transform an onnx to tensorrt ,this occured.
python: /root/gpgpu/MachineLearning/myelin/src/compiler/./ir/operand.h:166: myelin::ir::tensor_t*& myelin::ir::operand_t::tensor(): Assertion `is_tensor()' failed.

who can tell me how to solve this problem?

my environment :
cuda:10.2
tensorrt:8.2

**Comment by ttyio** [comment]
Hello @quintetoy  @yasserkhalil93 , sounds bug in TensorRT, could you share us the onnx the model to debug? thanks!

**Comment by Keysmis** [comment]
hello~, any update now? is it succeed?

**Comment by ttyio** [comment]
Hello @Keysmis , we did not get the repro to debug the issue. could you try 8.2 release? send us repro if it still failed. thanks!

**Comment by KeyKy** [comment]
I also get this error!

**Comment by aeoleader** [comment]
I am having the same issue with 8.2.2.1 version.

**Comment by ttyio** [comment]
@KeyKy  @aeoleader , could you provide us repro to debug the issue? thanks!

**Comment by aeoleader** [comment]
I have done a little bit of testing. It seems like a problem in the slice assignment. 

Repro steps:
1. Convert model to onnx
```python
def test_func(x):
    # works
    t = x
    # is_tensor error
    # t[0] = x[0]
    return t

class MyModule(torch.nn.Module):
    def __init__(self):
        super(MyModule, self).__init__()
    
    def forward(self, x):
        return test_func(x)

if __name__ == '__main__':
    model = MyModule()
    input_shape = (1, 16, 4, 224, 224)
    model.cpu().eval()
    model.to('cuda:0')
    input_tensor = torch.randn(input_shape).to('cuda:0')
    output_file = 'debug.onnx'
    torch.onnx.export(
    model,
    input_tensor,
    output_file,
    export_params=True,
    keep_initializers_as_inputs=True,
    verbose=False,
    opset_version=11)
```
2. Use the trtexec to compile with the following command:
```bash
trtexec --onnx=debug.onnx --saveEngine=debug.trt --best --buildOnly --workspace=4096 --verbose
```

**Comment by ttyio** [comment]
Thank you @aeoleader for the repro, I am able to repro the failure and create internal bug to track this.

**Comment by aeoleader** [comment]
@ttyio 
Can you also try [this](https://github.com/NVIDIA/TensorRT/issues/1708) as well? Constant padding seems broken as well.

**Comment by ttyio** [comment]
Hi @aeoleader , this `Assertion is_tensor()` failure will be fixed in next major release, thanks!

**Comment by pycoco** [comment]
> 

is it fix now? which trt version that i will use?

**Comment by pycoco** [comment]
i met the error:
Timing Runner: {ForeignNode[178...DequantizeLinear_40_dequantize_scale_node]} (Myelin)
trtexec: /root/gpgpu/MachineLearning/myelin/src/compiler/./ir/operand.h:166: myelin::ir::tensor_t*& myelin::ir::operand_t::tensor(): Assertion `is_tensor()' failed.

**Comment by ttyio** [comment]
This will be fixed in 8.4GA, thanks all

**Comment by wenqibiao** [comment]
> This will be fixed in 8.4GA, thanks all

Why NOT fix it in 8.2???

**Comment by Yebi1837** [comment]
I am facing the same error when trying to convert torch.index_add_() operator. Is there a work around for this with v8.2?

**Comment by wittenator** [comment]
Is there any known workaround for this issue? Since Nvidia does not update the CUDA version for the Nano, everybody with a Nano is stuck with this issue without the possibility to upgrade
